float64[3] params
---
bool success
